__init__.py

```diff
@@ -101,44 +101,68 @@ def __init__(self, bus):
-                continue
-            # For Display purposes, we want to show the engine name without the underscore or dash and capitalized all
-            plugin_display_name = engine.replace("_", " ").replace("-", " ").title()
-            for config in configs:
-                d = {"plugin_name": plugin_display_name,
-                     "display_name": config.get("display_name", " "),
-                     "offline": config.get("offline", False),
-                     "lang": lang,
-                     "engine": engine}
-                stt_opts.append(d)
-                self._stt_opts[hash_dict(d)] = config
-        return stt_opts
-
-    def get_tts_lang_options(self, lang, blacklist=None):
-        blacklist = blacklist or []
-        tts_opts = []
-        cfgs = get_tts_lang_configs(lang=lang, include_dialects=True)
-        for engine, configs in cfgs.items():
-            if engine in blacklist:
-                continue
-            # For Display purposes, we want to show the engine name without the underscore or dash and capitalized all
-            plugin_display_name = engine.replace("_", " ").replace("-", " ").title()
-            for voice in configs:
-                d = {"plugin_name": plugin_display_name,
-                     "display_name": voice.get("display_name", " "),
-                     "gender": voice.get("gender", " "),
-                     "offline": voice.get("offline", False),
-                     "lang": lang,
-                     'engine': engine}
-                tts_opts.append(d)
-                self._tts_opts[hash_dict(d)] = voice
-        return tts_opts
+    def get_stt_lang_options(self, lang, blacklist=None, preferred=None):
+        try:
+            blacklist = blacklist or []
+            stt_opts = []
+            cfgs = get_stt_lang_configs(lang=lang, include_dialects=True)
+            for engine, configs in cfgs.items():
+                if engine in blacklist:
+                    continue
+                # For Display purposes, we want to show the engine name without the underscore or dash and capitalized all
+                plugin_display_name = engine.replace("_", " ").replace("-", " ").title()
+                for config in configs:
+                    d = {"plugin_name": plugin_display_name,
+                        "display_name": config.get("display_name", " "),
+                        "offline": config.get("offline", False),
+                        "lang": lang,
+                        "engine": engine}
+                    stt_opts.append(d)
+                    self._stt_opts[hash_dict(d)] = config
+
+                # Sort the list for GUI to display the preferred STT engine first
+                # allow images to set a preferred engine
+                if preferred and preferred not in blacklist:
+                    stt_opts = sorted(stt_opts, key=lambda x: x["engine"] != preferred, reverse=False)
+
+            return stt_opts
+
+        except Exception as e:
```

Comment:
bump OPM and this wont be needed anymore

Reply:
i have updated to git master and it works but try and except will safe guard this from future fails and not hang on getting list, i think this should stay in there incase there  is any failure due to some bad config or something in OPM, chromium and mimic2 are anyways dependencies of the setup skill, so even if it fails to get some list we at least have one option to show as fallback and move on with the setup

Reply:
cleaned this up a bit, try except should still stay, and at least return an empty list if there is any error so its not stuck on the confirmation screen